src/Workflow/InstanceManager.cs

```diff
@@ -311,12 +311,22 @@ public static Guid Start(WorkflowHandlerBase workflowInstance)
         {
             try
             {
-                var wfApp = CreateWorkflowApplication(workflowInstance, WorkflowApplicationCreationPurpose.StartNew, null);
+                // forced reload/rebuild all pinned structures e.g. Content + it's fields for indexing.
+                var wfInstance = Node.Load<WorkflowHandlerBase>(workflowInstance.Id);
+                if (wfInstance.DisabledObservers != null)
```

Comment:
The two instances are mixed here. The if statement should check the workflowInstance parameter, not the local variable.